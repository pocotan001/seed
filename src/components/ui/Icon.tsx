import * as React from "react";
import styled from "styled-components";
import margin, { IMarginProps } from "~/components/styles/extends/margin";
import colors, { IColorKey } from "~/components/styles/theme/colors";
import { px } from "~/utils";

interface IIconProps extends IMarginProps {
  src: string;
  width?: number | string;
  height?: number | string;
  fill?: IColorKey;
  className?: string;
}

const Icon: React.SFC<IIconProps> = ({ src, className }) => (
  <span className={className} dangerouslySetInnerHTML={{ __html: src }} />
);

const StyledIcon = styled(Icon)`
  display: inline-block;
  vertical-align: middle;
  width: ${({ width }) => px(width!)};
  height: ${({ height }) => px(height!)};
  fill: ${({ fill }) => colors[fill!] || "currentColor"};
  ${margin};
`;

StyledIcon.defaultProps = {
  width: "1rem",
  height: "1rem"
};

export default StyledIcon;
