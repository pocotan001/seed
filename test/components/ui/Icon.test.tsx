import * as React from "react";
import * as renderer from "react-test-renderer";
import Icon from "~/components/ui/Icon";

const SVG_STRING = [
  `<svg width="100" height="100">`,
  `<circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="tomato" />`,
  `</svg>`
].join("");

describe("<Icon>", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Icon src={SVG_STRING} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
