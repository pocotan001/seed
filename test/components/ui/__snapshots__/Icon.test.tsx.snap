// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> should render correctly 1`] = `
<span
  className="sc-bdVaJa glhnSV"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg width=\\"100\\" height=\\"100\\"><circle cx=\\"50\\" cy=\\"50\\" r=\\"40\\" stroke=\\"black\\" stroke-width=\\"3\\" fill=\\"tomato\\" /></svg>",
    }
  }
/>
`;
